src/v/vlog.h

```diff
@@ -8,32 +8,8 @@
  * the Business Source License, use of this software will be governed
  * by the Apache License, Version 2.0
  */
-
 #pragma once
-
-#include <cstdint>
-
-namespace vlog_internal {
-constexpr int32_t basename_index(
-  const char* const path,
-  const int32_t index = 0,
-  const int32_t slash_index = -1) {
-    // NOLINTNEXTLINE
-    const char c = path[index];
-    if (c == '\0') {
-        return slash_index + 1;
-    }
-    if (c == '/' || c == '\\') {
-        return basename_index(path, index + 1, index);
-    }
-    return basename_index(path, index + 1, slash_index);
-}
-
-template<int32_t V>
-struct log_basename_start {
-    static constexpr const int32_t value = V;
-};
-} // namespace vlog_internal
+#include "utils/source_location.h"
```

Comment:
i wonder if we should just change this to `std::source_location` given our compiler versioning anyway?

Reply:
libc++ doesn't seem to have it yet